Makefile

```diff
@@ -213,6 +205,23 @@ tidy-check: gen_cmake ## Run clang-tidy
 			 \nFor detail information about the checks, please refer to https://clang.llvm.org/extra/clang-tidy/checks/list.html" \
 			 ; exit 1)
 
+.PHONY: list-tests
+list-tests: gen_cmake ## List all tests. This one is helpful to choose which test to run when calling `make single-test TEST_NAME=<test name>`
+	docker run  ${CONCORD_BFT_USER_GROUP} ${BASIC_RUN_PARAMS} \
+		${CONCORD_BFT_CONTAINER_SHELL} -c \
+		"cd ${CONCORD_BFT_BUILD_DIR} && \
+		ctest -N"
+
+# Test targets
+NUM_REPEATS?=1
```

Comment:
I already implemented repeats for apollo tests, you better use the implemented mechanism to avoid rerunning tests more times than you originally intended.
Just use the NUM_REPEATS environment variable in the with_bft_network function.
Gtest also supports repeats using the command line argument --gtest_repeat.

Reply:
I'm trying to remember why I went this way, as I've tried to use the internal capabilities of both testing framework and backed off (It's an old branch I forgot to push).

For gtest: you are correct, thank you for the comment: it would be beter to use --gtest_repeat, there is another flag  --gtest_break_on_failure. I will replace them both.

As for Apollo tests, there is no option currently to break on failure + the parameter is not visable in the Makefile.
I think it would be good to have them both - an option to break on failure when running inside container, and keep what i did. I don't think there could be a confusion as your parameter was intended to use form inside the container +  it is not reflected in the makefile, so it can never happen. Even if yes, we will get a multiplication, it's developer responsibility.

What do you think?

Reply:
Adding to the Apollo issue:
As I've said the repeat option you’ve added doesn’t support break on failure + it is not visible for the outside. Not all developers/simple users know how to run from inside the container.
We can have duplicated mechanisms or add this option with your loop + reflect it all the way to the Makefile, but I suggest doing this in another commit.